ocrd/ocrd/cli/workspace.py

```diff
@@ -19,25 +19,7 @@ class WorkspaceCtx():
 
     def __init__(self, directory, mets_url, mets_basename, automatic_backup):
         self.log = getLogger('ocrd.cli.workspace')
-        if mets_basename and mets_url:
-            raise ValueError("Use either --mets or --mets-basename, not both")
-        if mets_basename and not mets_url:
-            self.log.warning(DeprecationWarning("--mets-basename is deprecated. Use --mets/--directory instead"))
-        mets_basename = mets_basename if mets_basename else 'mets.xml'
-        if directory and mets_url:
-            directory = abspath(directory)
-            if not abspath(mets_url).startswith(directory):
-                raise ValueError("--mets has a directory part inconsistent with --directory")
-        elif not directory and mets_url:
-            if mets_url.startswith('http') or mets_url.startswith('https:'):
-                raise ValueError("--mets is an http(s) URL but no --directory was given")
-            directory = dirname(abspath(mets_url)) or getcwd()
-        elif directory and not mets_url:
-            directory = abspath(directory)
-            mets_url = join(directory, mets_basename)
-        else:
-            directory = getcwd()
-            mets_url = join(directory, mets_basename)
+        directory ,mets_url, mets_basename = resolve_mets_arguments(directory, mets_url, mets_basename, self.log)
```

Comment:
Wouldn't it be better to keep the `DeprecationWarning` here, since it only applies to the workspace CLI?

Reply:
The deprecation warning will still be raised, it's just been moved to `ocrd_utils.os.resolve_mets_arguments`:

```python
if mets_basename and mets_url:                                                                                           
    raise ValueError("Use either --mets or --mets-basename, not both")                                                   
elif not mets_basename and mets_url:                                                                                     
    mets_basename = Path(mets_url).name                                                                                  
elif not mets_basename and not mets_url:                                                                                 
    mets_basename = 'mets.xml'                                                                                           
else:                                                                                                                    
    (log.warning if log else print)(DeprecationWarning("--mets-basename is deprecated. Use --mets/--directory instead")) 
```

Reply:
Yes, but _shouldn't_ it remain here (since there is nothing to deprecate elsewhere)?

Reply:
56edc4c9 reintroduces the DeprecationWarning to `ocrd.cli.workspace` and replaces this call with an actual Python warning.

BTW I rue the day of thinking that `--mets-basename` was a good idea...